contracts/issuer/IssuerTopFacet.sol

```diff
@@ -59,6 +59,29 @@ contract IssuerTopFacet is AIssuerFacet {
     emit FastRegistered(fast);
   }
 
+  /**
+   * @notice Allows the unregistration of a given FAST diamond with this Issuer.
+   * @param fast The address of the FAST diamond to be unregistered.
+   * @notice Requires that the caller is a member of this Issuer.
+   * @notice Emits a `FastUnregistered` event.
+   */
+  function unregisterFast(address fast)
+    external
+    onlyMember(msg.sender) {
+      LibIssuer.Data storage s = LibIssuer.data();
+      string memory symbol = FastTokenFacet(fast).symbol();
+
+      // Disable transfers in the FAST.
+      FastTopFacet(fast).setTransfersDisabled(true);
+
+      // Remove the FAST from our lists.
+      s.fastSet.remove(fast, false);
+      delete s.fastSymbols[symbol];
+
+      // Emit!
+      emit FastUnregistered(fast);
```

Comment:
Do we want to also have a `reason` `string` in the event?

Reply:
Probably not on-chain... Maybe when we bake UI on top, we'll want to specify it on the backend.
If someone messes up, we probably don't want to say why publicly :)
But I'll double-check with @sankalp8i .

Reply:
Haha true!